src/test/java/ui/WeekViewTest.java

```diff
@@ -0,0 +1,63 @@
+package ui;
+
+import data.TaskManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import time.WeekView;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class WeekViewTest {
+
+    private WeekView weekView;
+    private TaskManager taskManagerMock;
+
+    @BeforeEach
+    public void setUp() {
+        LocalDate startOfWeek = LocalDate.of(2024, 3, 10); // March 10, 2024
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("MM/dd/yyyy");
+        weekView = new WeekView(startOfWeek, dateFormatter);
+        taskManagerMock = new TaskManager(); // You can create a mock TaskManager as needed
+    }
+
+    @Test
+    public void testNextWeek() {
+        LocalDate initialStartOfWeek = weekView.getStartOfWeek();
+        weekView.nextWeek();
+        LocalDate newStartOfWeek = initialStartOfWeek.plusWeeks(1);
+        assertEquals(newStartOfWeek, weekView.getStartOfWeek());
+    }
+
+    @Test
+    public void testPreviousWeek() {
+        LocalDate initialStartOfWeek = weekView.getStartOfWeek();
+        weekView.previousWeek();
+        LocalDate newStartOfWeek = initialStartOfWeek.minusWeeks(1);
+        assertEquals(newStartOfWeek, weekView.getStartOfWeek());
+    }
+
+    @Test
+    public void testNextMonth() {
```

Comment:
Perhaps testNextMonth() and testPreviousMonth() could be refactored into one method with an input with parameters that change previous or next